derivation.nix

```diff
@@ -15,15 +15,23 @@ let
 in
 {
   mkPnpmPackage =
-    { src
+    { workspace ? null
+    , components ? []
+    , src ? if (workspace != null && components != []) then workspace else null
     , packageJSON ? src + "/package.json"
-    , pnpmLockYaml ? src + "/pnpm-lock.yaml"
+    , componentPackageJSONs ? map (c: {
+        name = "${c}/package.json";
+        value = src + "/${c}" + "/package.json";
+      }) components
+    , pnpmLockYaml ? (if workspace == null then src else workspace) + "/pnpm-lock.yaml"
```

Comment:
It would be simpler to not attempt to support the case where `workspace != src`. If I make that assumption, I can drop this diff.

Reply:
Done.